plasmapy/plasma/grids.py

```diff
@@ -96,14 +168,13 @@ def grids(self):
                 self._grids = (pts0, pts1, pts2)
             else:
                 self._grids = (
-                    self.ds["ax0"] * self.unit0,
-                    self.ds["ax1"] * self.unit1,
-                    self.ds["ax2"] * self.unit2,
+                    self.ds["ax0"].data,
+                    self.ds["ax1"].data,
+                    self.ds["ax2"].data,
                 )
 
         return self._grids
 
-    # Note: may remove this function?
     @property
     def grid(self):
```

Comment:
There a couple subtle issues with how this method is currently written.

1. The method is actually returning a copy of the data contained in the `xarray.Dataset`.  This is not a problem in and of itself, but the docstring should be updated to specifically state this.  For example, `"Return a copy of the positional data as an (N, N, N, 3) array."`

   This is how the method currently operates.
   ```python
   >>> np.array_equal(grid.ds["ax0"], grid.grid[..., 0, 0, 0])
   True
   >>> grid.grid[0, 0, 0, 0] = 5.0
   >>> np.array_equal(grid.ds["ax0"], grid.grid[..., 0, 0, 0])
   False
   ```
2. Since the returned variable is only defined when `self._grid == None` or `self._grids == None`, this property will never get updated when the `xarray.Dataset` is modified.  There is some annoying overhead with this, but the array should probably be reconstructed every time the property is called.

Reply:
I'm aware of this issue: I was worried that the overhead in reconstructing these arrays would be costly in simulation loops. I'll change it to be the way you suggest for now. If it turns out to be a big performance hit, we can try to find another way?